### PR TITLE
[10.0][ADD] stock_valued_picking_report_triple_discount

### PR DESCRIPTION
**Pull request by luismontalba**
Stock Valued Picking Report Triple Discount
===================================
Add triple discount information to valued Delivery Slip report for partners where the check to avoid valued picking is set.

Configuration
-----------------
* Go to *Customers > (select one of your choice) > Sales & Purchases*.
* Set *Valued picking* field on.
* Go to *Sales > Configuration > Settings*.
* Set *Allow discounts on sales order lines* field on.

Usage
-------
To get the triple discount stock picking valued report:
* Create a Sale Order with stockable products a *Valued picking* able customer.
* Create different discounts in sale order lines.
* Confirm the Sale Order.
* Click on *Deliveries* stat button.
* Go to *Print > Delivery Slip*.

cc @Tecnativa

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Looks good. Can you achieve 100% coverage?

File: stock_picking_value_triple_discount/i18n/es.po [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+#. module: stock_valued_picking_triple_discount
+#: model:ir.ui.view,arch_db:stock_valued_picking_triple_discount.valued_report_picking
+msgid "<strong>Delivered Quantity</strong>"
+msgstr "<strong>Cantidad entregada</strong>"
+
+#. module: stock_valued_picking_triple_discount
+#: model:ir.ui.view,arch_db:stock_valued_picking_triple_discount.valued_report_picking
+msgid "<strong>Disc.2(%)</strong>"
+msgstr "<strong>Disc.2(%)</strong>"
```
Comment (pedrobaeza) [inline stock_picking_value_triple_discount/i18n/es.po]: This is not translated

File: stock_picking_value_triple_discount/models/stock_pack_operation.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        digits=dp.get_precision('Discount'),
+        string="Sale discount 2 (%)")
+    sale_discount3 = fields.Float(
+        compute='_compute_sale_order_line_fields',
+        digits=dp.get_precision('Discount'),
+        string="Sale discount 3 (%)")
+
+    @api.multi
+    def sale_lines_values(self, sale_lines):
+        discount_extra = (sale_lines.mapped('discount2') +
```
Comment (pedrobaeza) [inline stock_picking_value_triple_discount/models/stock_pack_operation.py]: Why do you need to compute again all the fields? Please make the other module inheritable using some hook methods for avoiding rewriting entire method.

File: stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="valued_report_picking" inherit_id="stock.report_delivery_document">
+        <xpath expr="//table[@t-if='o.pack_operation_ids']"
+               position="replace">
```
Comment (pedrobaeza) [inline stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml]: Never make a replace operation.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by luismontalba** [comment]
Changes done and 100% coverage reached.

**Comment by luismontalba** [comment]
ping @chienandalu , @pedrobaeza

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Please name the module correctly:  `stock_valued_picking_report_triple_discount`

File: stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="valued_report_picking" inherit_id="stock.report_delivery_document">
```
Comment (chienandalu) [inline stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml]: shouldn't the `inherit_id` be `stock_valued_picking_report.valued_report_picking`?

**Review by chienandalu — APPROVED** [review]
Tested on runbot :+1: Just a slight doubt:

File: stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml (hunk 4 above)
Comment (luismontalba) [reply stock_picking_value_triple_discount/report/stock_valued_picking_triple_discount.xml]: You are right, changes done

**Comment by pedrobaeza** [comment]
The module name, please...

**Comment by luismontalba** [comment]
Module name updated

**Comment by luismontalba** [comment]
Now the module is correctly renamed.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**